src/System.Text.Json/tests/Serialization/Stream.WriteTests.cs

```diff
@@ -258,6 +259,488 @@ public static async Task LargeJsonFile(int bufferSize)
             }
         }
+
+        [Theory]
+        [InlineData(1, true, true)]
+        [InlineData(1, true, false)]
+        [InlineData(1, false, true)]
+        [InlineData(1, false, false)]
+        [InlineData(10, true, true)]
+        [InlineData(10, true, false)]
+        [InlineData(10, false, true)]
+        [InlineData(10, false, false)]
+        [InlineData(100, true, true)]
+        [InlineData(100, true, false)]
+        [InlineData(100, false, true)]
+        [InlineData(100, false, false)]
+        [InlineData(1000, true, true)]
+        [InlineData(1000, true, false)]
+        [InlineData(1000, false, true)]
+        [InlineData(1000, false, false)]
+        public static async Task VeryLargeJsonFileTest(int payloadSize, bool ignoreNull, bool writeIndented)
+        {
+            List<Post> list = populateLargeObject(payloadSize);
+
+            JsonSerializerOptions options = new JsonSerializerOptions
+            {
+                IgnoreNullValues = ignoreNull,
+                WriteIndented = writeIndented
+            };
+
+            string json = JsonSerializer.Serialize(list, options);
+
+            // Sync case.
+            {
+                List<Post> deserializedList = JsonSerializer.Deserialize<List<Post>>(json, options);
+                Assert.Equal(payloadSize, deserializedList.Count);
+
+                string jsonSerialized = JsonSerializer.Serialize(deserializedList, options);
+                Assert.Equal(json, jsonSerialized);
+            }
+
+            // Async case.
+            using (var memoryStream = new MemoryStream())
+            {
+                await JsonSerializer.SerializeAsync(memoryStream, list, options);
+                string jsonSerialized = Encoding.UTF8.GetString(memoryStream.ToArray());
+                Assert.Equal(json, jsonSerialized);
+
+                memoryStream.Position = 0;
+                List<Post> deserializedList = await JsonSerializer.DeserializeAsync<List<Post>>(memoryStream, options);
+                Assert.Equal(payloadSize, deserializedList.Count);
+            }
+        }
+
+        [Theory]
+        [InlineData(1, true, true)]
+        [InlineData(1, true, false)]
+        [InlineData(1, false, true)]
+        [InlineData(1, false, false)]
+        [InlineData(2, true, true)]
+        [InlineData(2, true, false)]
+        [InlineData(2, false, true)]
+        [InlineData(2, false, false)]
+        [InlineData(4, true, true)]
+        [InlineData(4, true, false)]
+        [InlineData(4, false, true)]
+        [InlineData(4, false, false)]
+        [InlineData(8, true, true)]
+        [InlineData(8, true, false)]
+        [InlineData(8, false, true)]
+        [InlineData(8, false, false)]
+        [InlineData(16, true, true)]
+        [InlineData(16, true, false)]
+        [InlineData(16, false, true)]
+        [InlineData(16, false, false)]
+        public static async Task DeepJsonFileTest(int depthFactor, bool ignoreNull, bool writeIndented) {
```

Comment:
Style: Should be Allman braces, not K&R braces.

```suggestion
        public static async Task DeepJsonFileTest(int depthFactor, bool ignoreNull, bool writeIndented)
        {
```